src/system.c

```diff
@@ -1462,7 +1462,20 @@ Int8 SyNanosecondsSinceEpoch()
 {
   Int8 res;
 
-#if defined(HAVE_CLOCK_GETTIME) && defined(CLOCK_MONOTONIC)
+#if defined(SYS_IS_DARWIN)
+  UInt8 nanosecs;
+
```

Comment:
Is there a reason you are using `UInt8` here, (and not `res`?) and returning it directly, implicitly converting to `Int8`?